Source/Modules/python.cxx

```diff
@@ -854,7 +863,14 @@ class PYTHON:public Language {
       Printv(f_shadow, "del version_info\n", NULL);
 
       if (builtin) {
-	Printf(f_shadow, "from %s import *\n", module);
+        /*
+         * Python3 removes relative imports.  So 'from _foo import *'
+         * will only work for non-package modules.
+         */
+        Printv(f_shadow, "if __name__.rpartition('.')[0] != '':\n", NULL);
+        Printf(f_shadow, tab4 "from .%s import *\n", module);
```

Comment:
>>>>> " " == Pawe� Tomulik <notifications@github.com> writes:

    >> @@ -854,7 +863,14 @@ class PYTHON:public Language {
    >> Printv(f_shadow, "del version_info\n", NULL);
    >> 
    >> if (builtin) { - Printf(f_shadow, "from %s import *\n",
    >> module); + /* + * Python3 removes relative imports.  So 'from
    >> _foo import *' + * will only work for non-package modules.  +
    >> */ + Printv(f_shadow, "if __name__.rpartition('.')[0] !=
    >> '':\n", NULL); + Printf(f_shadow, tab4 "from .%s import *\n",
    >> module);

     > Maybe: ```c++ Printv(f_shadow, tab4 "from %s%s import *\n",
     > (py3 ? "." : ""), module ) ``` for py2 backward compatibility?

  Thanks for the tip on where to locate this!   At first glance this
looks like a more straight forward solution than the one I committed.
The down side to it is that it will not work if the module is built
for python3 and not placed into a package.  The shadow file will still
have 'from ._foo import *' in it and this will fail because _foo.so is
in the global namespace.

Mike